### PR TITLE
Add new filters to replace default/fixed commission rates

### PR DESCRIPTION
**Pull request by raphaelkross**
Adds two new filters that allow plugins overwrite the percentage/fixed values when calculating the commission amount.

### CODE REVIEW

**Comment by itzmekhokan** [comment]
@raphaelkross, thanks for contribution but it seems not a relevant PR.
You are actually trying to filter default something which is manageable by backend settings. Not only that as per WCMp flows commission rates respect follows priorities - 
Product <-- Product Category <-- Vendor <-- Default
So default commission rates will always be overridden by first priority wise commission rates although we use the above patch.
You can control WCMp final commission rates by follows filter - 
`apply_filters('wcmp_get_commission_amount', $commission, $product_id, $vendor->term_id, $variation_id, $item_id, $order);`

**Comment by raphaelkross** [comment]
Hi @itzmekhokan the idea behind this is that I do not want to re-do the calculation that WCMP already does, but customize the parameters used.

We are with a case use where we have multiple locations (Point of Sales) and according to this the percentages are modified - the commission calc is correct but the parameters should be customized, therefore I've added a couple extra filters - to enable this extra control without having to mess with the way WCMP calculates the commission.

**Comment by itzmekhokan** [comment]
@raphealkross, I think you got confused with the filter name and used function name. Actually it's a typo. If you checked the calculate commission functionalities, you can understand that the above shared filter only returns commission rates with parameters 'commission_val' & 'commission_fixed'. In your case default_percentage are store in commission_val and fixed store in commission_fixed. Basically if you check the function named `get_commission_amount()` again Im saying it's a typo it should be 'get_commission_rates()', you will understand that using the above filter you can do the modification respecting the point of sales concept. Where array data commission_val can be default percentage or product commission percentage or product category commission percentage or vendor commission percentage value and commission_fixed can be default fixed or all the fixed commission or fixed with quality values.
 
So, basically use the above filter and return your modified percentage and fixed values by checking you point of sales concept. Rest WCMp itself will calculate the commission respecting you filtered values.

Do let me know if you have some queries regarding this.

**Comment by raphaelkross** [comment]
Gotcha! Makes sense - really got confused by the filter name!

Thanks for clarifying.

**Comment by itzmekhokan** [comment]
@raphealkross I'm closing the PR.
